### ISSUE TITLE
Assistant quickfilters issue with Preception.

### ISSUE THREAD

**Issue by Redback813** [RealRaven2000/quickFilters#60]
I like to thank you for this add-on for thunderbird, I have notice that there could be an issue with this the "Assistant quickfilters".
 The first issue is when one comes to the final setup of a filter for whatever reason the end user might cancel the filter and there no issue however the msg are still copied or moved over to the destination folder even when the filter been cancelled, which the user would think the process of copy/move of msg would stop as well. Can this be corrected?

**Comment by RealRaven2000** [comment]
yes if you cancel the filter dialog when certain a new filter through assistant the expectation is that it is not being stored.  maybe its a side effect of the new sort search terms button - we're you using that before hitting cancel?

**Comment by RealRaven2000** [comment]
also if you move the mail and the assistant is triggered, any cancelling is not expected to undo the move.  the assistant just means to help automate the move (creating the filter) not to undo it. you can still use the undo button for that

**Comment by Redback813** [comment]
> yes if you cancel the filter dialog when certain a new filter through assistant the expectation is that it is not being stored. maybe its a side effect of the new sort search terms button - we're you using that before hitting cancel?

No to this question, it was a start up filter creation and I only modified the section pertaining to junk status and copy instead of move.

**Comment by RealRaven2000** [comment]
I still don't understand fully.  also if there are 2 issues can you please break them up into separate items? it is hard to fix multiple problems in the same issue...

>  the msg are still copied or moved over to the destination folder even when the message filter been cancelled

what you are saying is the filter is executed (applied to your inbox) when you click run filter or if it is configured to run automatically? that's how the run filter command is designed in Tb - it is applied on everything in the folder.

**Comment by RealRaven2000** [comment]
> The first issue is the final setup of the message filter for whatever reason the end user might cancel the message filter and there no issue however the msg are still copied or moved over to the destination folder even when the message filter been cancelled, which the user would think the process of copy/move of msg would stop as well

reading this part again - this one is definitely _by design_ - quickFilters cannot keep track of your message move in this case. Once you move the message, it assumes you want the message in the new location, even if you cancel filter creation. However the "undo" command on Thunderbird Edit menu might still be available to undo moving the message.

**Comment by RealRaven2000** [comment]
> The second issue is after completing a message filter setup and proceeds to the message filter processing job, for some unknown reason the copy/move process encompasses all msg including those not part of the message filter within a folder, regardless of the message filter the process still copy/move what not part of the message filter.

Still do not fully understand. You can move one or multiple messages and depending on the template quickFilters may use the first or all moved mails ("Group" template) for generating search conditions and filling the target folder item. If you then run the new filter on folder all search terms are applied to all remaining emails in the folder (typically the inbox). What exactly do you expect instead? It cannot apply them to anything else as you have already moved the original messages.

**Comment by RealRaven2000** [comment]
Last question, what is the meaning of the term "Preception"? I haven't found any translation for this maybe I am just thick (native German speaker here) ... I don't think I have heard the term before.

**Comment by Redback813** [comment]
To your question on the word “perception”. My mistake on the spelling!

 

perception | pəˈsɛpʃ(ə)n | noun [mass noun] 1 the ability to see, hear, or become aware of something through the senses: the normal limits to human perception. • awareness of something through the senses: the perception of pain. • Psychology & Zoology the neurophysiological processes, including memory, by which an organism becomes aware of and interprets external stimuli. 2 the way in which something is regarded, understood, or interpreted: Hollywood's perception of the tastes of the American public | [count noun] :  we need to challenge many popular perceptions of old age. • intuitive understanding and insight: ‘He wouldn't have accepted,’ said my mother with unusual perception. 

 

 

“maybe I am just thick (native German speaker here)” Nope and don’t sell ya self short sport.

 

Cheers

Andrew

 

 

From: RealRaven2000 ***@***.***>
Reply to: RealRaven2000/quickFilters ***@***.***>
Date: Wednesday, 21 April 2021 at 10:32 pm
To: RealRaven2000/quickFilters ***@***.***>
Cc: Redback813 ***@***.***>, Author ***@***.***>
Subject: Re: [RealRaven2000/quickFilters] Assistant quickfilters issue with Preception. (#60)

 

Last question, what is the meaning of the term "Preception"? I haven't found any translation for this maybe I am just thick (native German speaker here) ... I don't think I have heard the term before.

**Comment by RealRaven2000** [comment]
> > yes if you cancel the filter dialog when certain a new filter through assistant the expectation is that it is not being stored. maybe its a side effect of the new sort search terms button - we're you using that before hitting cancel?
> 
> No to this question, it was a start up filter creation and I only modified the section pertaining to junk status and copy instead of move.

ok, so
1) you moved messages
2) you created a filter that changes to copy
3) you cancel
4) you expect the original messages to be moved back?

I am probably wrong on at least one of these, it is not clear what you did and what you expect quickFilters to do. quickFilters can define filters and can cancel that. It does not keep track of messages, only reacts to the event of message counts changing in a folder.
